compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java

```diff
@@ -724,11 +724,11 @@ public static BLangMatchExpression createMatchExpression(BLangExpression expr) {
         return matchExpr;
     }
 
-    public static BLangFieldBasedAccess createFieldAccessExpr(BLangAccessibleExpression varRef, BLangIdentifier field) {
+    public static BLangFieldBasedAccess createFieldAccessExpr(BLangExpression varRef, BLangIdentifier field) {
         return createFieldAccessExpr(varRef, field, false);
     }
 
-    public static BLangFieldBasedAccess createFieldAccessExpr(BLangAccessibleExpression varRef, BLangIdentifier field,
+    public static BLangFieldBasedAccess createFieldAccessExpr(BLangExpression varRef, BLangIdentifier field,
```

Comment:
Why are we changing these?

Reply:
Reason to change that is, in `BLangAccessExpression` class type of `expr` field is `BLangExpression`.

Reply:
But IINM, right now it holds a `BLangAccessibleExpression`. So ideally whatever calls this should be passing a `BLangAccessibleExpression`.

Reply:
Here I used `BLangExpression` type `varRef` to create  `FieldAccessExpr`. Isn't it correct?
https://github.com/ballerina-platform/ballerina-lang/blob/5d1f6c2a2a6098c33380a9bda60806148760407c/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L7399

Reply:
The `varargRef` variable you are using here is actually a `BLangSimpleVarRef`, right? If you change the type of the param to `BLangSimpleVarRef` in `reorderNamedArgs`, this could work without any other changes.

Reply:
Changed as we discuss.